byterun/floats.c

```diff
@@ -530,6 +650,49 @@ CAMLprim value caml_log1p_float(value f)
   return caml_copy_double(caml_log1p(Double_val(f)));
 }
 
+#ifndef HAS_C99_FLOAT_OPS
+static inline double simple_erf(double x)
+{
+  double a1 =  0.254829592, a2 = -0.284496736, a3 = 1.421413741,
+         a4 = -1.453152027, a5 =  1.061405429, p  = 0.3275911,
+         t, y;
+
+  int sign = (x >= 0) ? 1 : -1;
+  x = fabs(x);
+  t = 1.0 / (1.0 + p * x);
+  y = 1.0 - (((((a5 *t  + a4) * t) + a3) * t + a2) * t + a1) * t * exp(-x * x);
```

Comment:
> I've just tried it out, and there is no appreciable difference in performance between exp and expf on my platform.

Well, on my linux machine (glibc 2.19) with a Core i3-2120 @ 3.3GHz, I did the following micro-bench:

```
int main () {
  double sum = 0;
  for(double i = 0; i < 10; i+=1E-7) {
    sum += exp(i);
  }
  printf("%f", sum);
  return 0;
}
```

With `exp`: 2.338s, with `expf`: 1.325s. That's fairly appreciable.

Anyway, I am not saying that this is what we should do. I am just saying that this might be worth trying, just to see how this affects the  precision/performance trade-off.